File: laravel/app/Models/Task.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Task extends Model
{
    protected $fillable = [
        'wedding_id', 'phase_id', 'title', 'description', 'type',
        'category', 'priority', 'due_date', 'completed', 'completed_at',
        'order', 'notes',
    ];

    protected $casts = [
        'due_date' => 'date',
        'completed' => 'boolean',
        'completed_at' => 'datetime',
    ];

    public function wedding(): BelongsTo
    {
        return $this->belongsTo(Wedding::class);
    }

    public function phase(): BelongsTo
    {
        return $this->belongsTo(Phase::class);
    }

    public function isExecution(): bool
    {
        return $this->type === 'execution';
    }

    public function isInput(): bool
    {
        return $this->type === 'input';
    }

    public function isOverdue(): bool
    {
        return !$this->completed && $this->due_date && $this->due_date->isPast();
    }

    public function getDaysRemainingAttribute(): ?int
    {
        if (!$this->due_date) return null;
        return (int) now()->diffInDays($this->due_date, false);
    }
}

File: flutter_app/lib/data/models/task.dart
import 'package:json_annotation/json_annotation.dart';

part 'task.g.dart';

@JsonSerializable()
class Task {
  final int id;
  @JsonKey(name: 'wedding_id')
  final int weddingId;
  @JsonKey(name: 'phase_id')
  final int phaseId;
  final String title;
  final String? description;
  final String type;
  final String category;
  final String priority;
  @JsonKey(name: 'due_date')
  final DateTime? dueDate;
  final String? notes;
  @JsonKey(name: 'is_completed')
  final bool isCompleted;
  @JsonKey(name: 'completed_at')
  final DateTime? completedAt;
  @JsonKey(name: 'sort_order')
  final int sortOrder;

  const Task({
    required this.id,
    required this.weddingId,
    required this.phaseId,
    required this.title,
    this.description,
    required this.type,
    required this.category,
    required this.priority,
    this.dueDate,
    this.notes,
    required this.isCompleted,
    this.completedAt,
    required this.sortOrder,
  });

  factory Task.fromJson(Map<String, dynamic> json) => _$TaskFromJson(json);

  Map<String, dynamic> toJson() => _$TaskToJson(this);
}
